R/cal-estimate.R

```diff
@@ -0,0 +1,405 @@
+                                                  times = 10,
+                                                  ...) {
+  truth <- enquo(truth)
+
+  levels <- truth_estimate_map(.data, !!truth, {{ estimate }})
+
+  if (length(levels) == 2) {
+    log_model <- cal_isoreg_boot(
+      .data = .data,
+      truth = !!truth,
+      estimate = levels[[1]],
+      times = times,
+      ...
+    )
+
+    res <- as_binary_cal_object(
+      estimate = log_model,
+      levels = levels,
+      truth = !!truth,
+      method = "Isotonic Bootstrapped",
+      additional_class = "cal_estimate_isotonic_boot"
+    )
+  } else {
+    stop_multiclass()
+  }
+
+  res
+}
+
+#------------------------- Estimate implementation -----------------------------
+#------------------------------ >> Logistic ------------------------------------
+cal_estimate_logistic_impl <- function(.data,
+                                       truth = NULL,
+                                       estimate = dplyr::starts_with(".pred_"),
+                                       type,
+                                       model,
+                                       method,
+                                       additional_class,
+                                       ...) {
+  truth <- enquo(truth)
+
+  levels <- truth_estimate_map(.data, !!truth, {{ estimate }})
+
+  if (length(levels) == 2) {
+    log_model <- cal_model_impl(
+      .data = .data,
+      truth = !!truth,
+      estimate = levels[[1]],
+      run_model = model,
+      ...
+    )
+
+    res <- as_binary_cal_object(
+      estimate = log_model,
+      levels = levels,
+      truth = !!truth,
+      method = method,
+      additional_class = additional_class
+    )
+  } else {
+    stop_multiclass()
+  }
+
+  res
+}
+
+cal_model_impl <- function(.data, truth, estimate, run_model, ...) {
+  truth <- ensym(truth)
+
+  if (run_model == "logistic_spline") {
+    f_model <- expr(!!truth ~ s(!!estimate, k = 10))
+    init_model <- mgcv::gam(f_model, data = .data, family = "binomial", ...)
+    model <- butcher::butcher(init_model)
+  }
+
+  if (run_model == "glm") {
+    f_model <- expr(!!truth ~ !!estimate)
+    init_model <- glm(f_model, data = .data, family = "binomial", ...)
+    model <- butcher::butcher(init_model)
+  }
+
+  model
+}
+
+#------------------------------ >> Isotonic ------------------------------------
+cal_isoreg_dataframe <- function(.data,
+                                 truth,
+                                 estimate,
+                                 sampled = FALSE,
+                                 ...) {
+  sort_data <- dplyr::arrange(.data, !!estimate)
+
+  if (sampled) {
+    sort_data <- dplyr::slice_sample(
+      .data = sort_data,
+      prop = 1,
+      replace = TRUE
+    )
+  }
+
+  x <- dplyr::pull(sort_data, !!estimate)
+
+  truth <- dplyr::pull(sort_data, {{ truth }})
+  y <- as.integer(as.integer(truth) == 1)
+
+  model <- isoreg(x = x, y = y)
```

Comment:
whenever you can, namespace functions.

Reply:
People might not run these functions on sock clusters,  but every non-namespaced function will fail because it can't be found in sock workers.